target/linux/mvebu/files-6.6/arch/arm64/boot/dts/marvell/armada-7040-rb5009.dts

```diff
@@ -0,0 +1,384 @@
+	/* This port is connected to 88E6393X switch */
+	status = "okay";
+	phy-mode = "10gbase-r";
+	phys = <&cp0_comphy2 0>;
+	managed = "in-band-status";
+	nvmem-cells = <&macaddr_hard 0>;
+	nvmem-cell-names = "mac-address";
+};
+
+&cp0_mdio {
+	status = "okay";
+
+	switch@0 {
+		/* Actual device is MV88E6393X */
+		compatible = "marvell,mv88e6190";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+		/* LED config is lost if switch is reset */
+		//reset-gpios = <&cp0_gpio2 2 GPIO_ACTIVE_HIGH>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				label = "cpu";
+				ethernet = <&cp0_eth0>;
+				phy-mode = "10gbase-r";
+				managed = "in-band-status";
+			};
+
+			port@1 {
+				reg = <1>;
+				label = "p8";
+				phy-handle = <&switch0phy1>;
+				nvmem-cells = <&macaddr_hard 7>;
+				nvmem-cell-names = "mac-address";
+			};
+
+			port@2 {
+				reg = <2>;
+				label = "p7";
+				phy-handle = <&switch0phy2>;
+				nvmem-cells = <&macaddr_hard 6>;
+				nvmem-cell-names = "mac-address";
+			};
+
+			port@3 {
+				reg = <3>;
+				label = "p6";
+				phy-handle = <&switch0phy3>;
+				nvmem-cells = <&macaddr_hard 5>;
+				nvmem-cell-names = "mac-address";
+			};
+
+			port@4 {
+				reg = <4>;
+				label = "p5";
+				phy-handle = <&switch0phy4>;
+				nvmem-cells = <&macaddr_hard 4>;
+				nvmem-cell-names = "mac-address";
+			};
+
+			port@5 {
+				reg = <5>;
+				label = "p4";
+				phy-handle = <&switch0phy5>;
+				nvmem-cells = <&macaddr_hard 3>;
+				nvmem-cell-names = "mac-address";
+			};
+
+			port@6 {
+				reg = <6>;
+				label = "p3";
+				phy-handle = <&switch0phy6>;
+				nvmem-cells = <&macaddr_hard 2>;
+				nvmem-cell-names = "mac-address";
+			};
+
+			port@7 {
+				reg = <7>;
+				label = "p2";
+				phy-handle = <&switch0phy7>;
+				nvmem-cells = <&macaddr_hard 1>;
+				nvmem-cell-names = "mac-address";
+			};
+
+			port@9 {
+				reg = <9>;
+				label = "p1";
+				phy-mode = "sgmii";
+				phy-handle = <&qca8081>;
+				managed = "in-band-status";
+				nvmem-cells = <&macaddr_hard 0>;
+				nvmem-cell-names = "mac-address";
+			};
+
+			port@a {
+				reg = <10>;
+				label = "sfp";
+				phy-mode = "10gbase-r";
+				managed = "in-band-status";
+				sfp = <&sfp>;
+				nvmem-cells = <&macaddr_hard 8>;
+				nvmem-cell-names = "mac-address";
+			};
+		};
+
+		mdio {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			switch0phy1: switch0phy1@1 {
+				reg = <0x1>;
+			};
+
+			switch0phy2: switch0phy2@2 {
+				reg = <0x2>;
+			};
+
+			switch0phy3: switch0phy3@3 {
+				reg = <0x3>;
+			};
+
+			switch0phy4: switch0phy4@4 {
+				reg = <0x4>;
+			};
+
+			switch0phy5: switch0phy5@5 {
+				reg = <0x5>;
+			};
+
+			switch0phy6: switch0phy6@6 {
+				reg = <0x6>;
+			};
+
+			switch0phy7: switch0phy7@7 {
+				reg = <0x7>;
+			};
+		};
+
+		mdio1 {
+			compatible = "marvell,mv88e6xxx-mdio-external";
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			qca8081: qca8081@0 {
+				reg = <0>;
+			};
```

Comment:
No idea how _valid_ it is but if I `active-low` the LED, I seem to get matching behaviour with green and amber (does not come on) between RouterBoot, u-boot, and OpenWrt (but appears to spend some seconds inverted while booting before this gets applied).
```
			qca8081: qca8081@0 {
				compatible = "ethernet-phy-id004d.d101";
				reg = <0>;

				leds {
					#address-cells = <1>;
					#size-cells = <0>;

					led@0 {
						reg = <0>;
						color = <LED_COLOR_ID_GREEN>;
						function = LED_FUNCTION_LAN;
						default-state = "keep";
						active-low;
					};


					led@1 {
						reg = <1>;
						color = <LED_COLOR_ID_AMBER>;
						function = LED_FUNCTION_LAN;
						default-state = "keep";
						active-low;
					};
				};
			};
```

Reply:
This should be valid, that is why Ansuel added the active-low cause some actually use it like that.

I need to get around to RB5009 (Hopefully tomorrow) to get this sorted out.

Reply:
is it related to rb5009 sfp port led never lighting up? i tried sfp+ dac, 1g, 2.5g and 10g copper modules - no light come up but module is detected by openwrt

Reply:
> is it related to rb5009 sfp port led never lighting up? i tried sfp+ dac, 1g, 2.5g and 10g copper modules - no light come up but module is detected by openwrt

No, not related to SFP LED. You need to define a trigger for it, see `cat /sys/class/leds/green:sfp/trigger`.
Example:
```base
root@OpenWrt-rb5009:~# grep -A10 sfp /etc/config/system 
config led 'sfp'
        option name 'SFP'
        option sysfs 'green:sfp'
        option trigger 'netdev'
        option dev 'sfp'
        option mode 'tx rx'
        option interval '50'
```
```base
/etc/init.d/led restart
```

Reply:
thanks, that solves sfp port.

been playing around with rb5009 and noticed 2.5gbe port light always being on without cable being plugged in. light gets proper triggering when there is activity. is that the issue with light above? :)

i have two rb5009 and both are the same.
![photo_2024-07-11_18-50-31](https://github.com/openwrt/openwrt/assets/450390/fb3e55af-0dc5-49d9-97ec-272bf67eea49)

Reply:
> been playing around with rb5009 and noticed 2.5gbe port light always being on without cable being plugged in.

That issue has been addressed with latest commits.